tests/unit_tests/tabular_test.py

```diff
@@ -16,12 +18,28 @@ def test_read_diann_removes_metadata_and_sets_index():
     # Expected DataFrame
     expected = pd.DataFrame(
         {
-            "Intensity.Sample1": [1000, 1500],
-            "Intensity.Sample2": [2000, 2500],
-        }, 
-        index=["P12345", "P23456"]
+            # The real diann data has float values in the intensities.
+            "Intensity.Sample1": [1000.0, 1500.0],
+            "Intensity.Sample2": [2000.0, 2500.0],
+        },
+        index=["P12345", "P23456"],
     )
     expected.index.name = "Protein"
 
     result = read_diann(mock_data)
     assert_frame_equal(result, expected)
```

Comment:
@ltatka this is the test I am talking about, the frame contents are the same except for the fact that its always floats, whilst before they could be a mix of ints and floats depending on the input data.